src/token.ts

```diff
@@ -1,8 +1,393 @@
+
+    constructor(line: string) {
+        const match = line.replace(regex.title_page, '\n$1').split(regex.splitter).reverse();
+        this.tokens = match.reduce(
+            (previous, item) => new TitlePageToken(item).addTo(previous)
+        , [])
+    }
+
+    addTo(tokens: Token[]): Token[] {
+        return [...tokens, ...this.tokens]
+    }
+
+    static matchedBy(line: string): boolean {
+        return regex.title_page.test(line)
+    }
+}
+
+export class TitlePageToken implements Token {
+    readonly type: string
+    readonly is_title = true
+    readonly text: string
+
+    constructor(item: string) {
+        const pair = item.replace(regex.cleaner, '').split(/\:\n*/);
+        this.type = pair[0].trim().toLowerCase().replace(' ', '_')
+        this.text = pair[1].trim()
+    }
+
+    addTo(tokens: Token[]): Token[] {
+        return [...tokens, this]
+    }
+}
+
+export class SceneHeadingToken implements Token {
+    readonly type = 'scene_heading'
+    readonly text: string
+    readonly scene_number: string
+    readonly too_short: boolean
+
+    constructor(line: string) {
+        const match = line.match(regex.scene_heading)
+        this.text = match[1] || match[2];
+        this.too_short = isTooShort(this.text)
+
+        const meta: RegExpMatchArray = this.text.match(regex.scene_number);
+        if (meta) {
+            this.scene_number = meta[2];
+            this.text = this.text.replace(regex.scene_number, '');
+        }
+    }
+
+    addTo(tokens: Token[]): Token[] {
+        if (this.too_short) {
+            return tokens;
+        }
+
+        return [...tokens, this]
+    }
+
+    static matchedBy(line: string): boolean {
+        return regex.scene_heading.test(line)
+    }
+}
+
+export class CenteredToken implements Token {
+    readonly type = 'centered'
+    readonly text: string
+
+    constructor(line: string) {
+        const match = line.match(regex.centered)
+        this.text = match[0].replace(/>|</g, '')
+    }
+
+    addTo(tokens: Token[]): Token[] {
+        return [...tokens, this]
+    }
+
+    static matchedBy(line: string): boolean {
+        return regex.centered.test(line)
+    }
+}
+
+export class TransitionToken implements Token {
+    readonly type = 'transition'
+    readonly text: string
+
+    constructor(line: string) {
+        const match = line.match(regex.transition)
+        this.text = match[1] || match[2]
+    }
+
+    addTo(tokens: Token[]): Token[] {
+        return [...tokens, this]
+    }
+
+    static matchedBy(line: string): boolean {
+        return regex.transition.test(line)
+    }
+}
+
+export class DialogueBlock implements Block {
+    readonly tokens: Token[] = []
+    readonly dual: boolean
+    readonly too_short: boolean
+    
+    constructor(line: string, dual: boolean) {
+        const match = line.match(regex.dialogue);
+
+        let name = match[1] || match[2];
+        this.too_short = isTooShort(name) && !line.startsWith('@')
```

Comment:
This one also seems to be unused but to be honest, I'm not sure if the `isTooShort` method is really needed here either.

What I've noticed is that main seems to turn a one character forced name into an action. 

Was this the intention of commit [757938b](https://github.com/jonnygreenwald/fountain-js/pull/11/commits/757938b17eb90dfe6de8d698f3740c64863b90c4) in the first place?

Considering the [Fountain Spec](https://fountain.io/syntax/#character) says:

> Character names must include at least one alphabetical character.

I think that this line can be removed but am open to why it should stay even though currently it seems to not do anything.

Reply:
You got it! That was the primary reason for that commit.

I left the logic there so there weren't any unintended side effects from changing it.

I have no problem removing it entirely if you don't!

Reply:
> I have no problem removing it entirely if you don't!

Yeah, sounds good to me. Thank you!